### PR TITLE
chore: update api test suite and include it in run test

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

It looks like a lot, but the main thing this PR does is fix up the supertest tests so that the server does not need to be running in the background before we run the tests.

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
_i want to get in a few more prs, before we activate tests, and they hinder my bugs_

File: .github/workflows/node.js-tests-upcoming.yml [hunk 1]
```diff
@@ -68,6 +68,7 @@ jobs:
       fail-fast: false
       matrix:
         node-version: [18.x]
+        mongodb-version: [6.0]
```
Comment (raisedadead) [inline .github/workflows/node.js-tests-upcoming.yml:71]: Does this have to be in the matrix? I do not see us testing in multiple versions of MongoDB, and it means we have to update every open PR for the updated check status this entails.

**Review by raisedadead — CHANGES_REQUESTED** [review]

File: .github/workflows/node.js-tests-upcoming.yml (hunk 1 above)
Comment (raisedadead) [reply .github/workflows/node.js-tests-upcoming.yml:71]: An alternative would be to use an exact match in the task stanza as an environment variable. I do not know your reasoning here, so I'll defer to what you think is better without using the matrix.

Comment (ojeytonwilliams) [reply .github/workflows/node.js-tests-upcoming.yml:71]: Nope, I just assumed you'd want to! I'll take them out.

**Review by raisedadead — APPROVED** [review]

File: api/app.ts [hunk 2]
```diff
@@ -18,38 +25,26 @@ import {
...
-}).withTypeProvider<TypeBoxTypeProvider>();
-
-export type FastifyInstanceWithTypeProvider = typeof fastify;
+export type FastifyInstanceWithTypeProvider = FastifyInstance<
+  RawServerDefault,
+  RawRequestDefaultExpression,
+  RawReplyDefaultExpression,
+  FastifyBaseLogger,
+  TypeBoxTypeProvider
+>;
```
Comment (ShaunSHamilton) [inline api/app.ts:38]: Just so we are on the same page (out of interest): Why this?

File: api/server.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    },
+    level: 'debug'
+  },
+  // TODO: is this the right level for production or should we use 'error'?
+  production: { level: 'fatal' },
+  test: undefined
+};
+
+const start = async () => {
+  const fastify = await build({ logger: envToLogger[NODE_ENV] });
```
Comment (ShaunSHamilton) [inline api/server.ts:22]: We won't tell Mrugesh about this one 🤫

File: api/app.ts (hunk 2 above)
Comment (ojeytonwilliams) [reply api/app.ts:38]: Limitations of my TypeScript skills, basically.  We want to correctly type the `fastify` instance, but we're using the `TypeBoxTypeProvider`, so we can't just use `FastifyInstance`'s defaults as they include `FastifyTypeProviderDefault`.

Before we could just use `typeof fastify`, but now `fastify` is created in a function and you can't export from functions.

Anyway, if there's a neater way to generate this, I don't know it.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Pull request merged**